node/src/testing/multi_stage_test_reactor/test_chain.rs

```diff
@@ -26,6 +26,10 @@ use crate::{
     NodeRng,
 };
 
+const ERA_TO_JOIN: u64 = 3;
+const MERKLE_TREE_HASH_ACTIVATION: u64 = ERA_TO_JOIN + 1;
+const MERKLE_TREE_HASH_ACTIVATION_FOR_ARCHIVAL_SYNC: u64 = ERA_TO_JOIN;
```

Comment:
Why use a different value for the archival sync test?

It also seems a bit dangerous; we'll end up accidentally using these in the wrong test. Should these constants be local to the individual tests instead?

Reply:
Inspired by this comment I changed the approach to the tests as follows:
- set the `MERKLE_TREE_HASH_ACTIVATION` individually for each test that involves this parameter
- if the test allows for the activation point to be arbitrarily chosen, I use random value
- otherwise, I execute the test using **both** legacy blocks and merkle hashed blocks

`block` tests are updated here: https://github.com/casper-network/casper-node/pull/2531/commits/f9e0ba78556fb1b2856974b241d758887da882f7

`storage` tests are updated here: https://github.com/casper-network/casper-node/pull/2531/commits/1c233d024bc5aa05a9e59a77a97a17e615fe7050

Tests still to be updated/reviewed:
- deploy_acceptor
- gossiper
- linear_chain
- linear_chain_sync
- multi_stage_test_reactor

Please not the in the two commits above there is some temporary code (like: `random_with_specifics_1()`) which will be removed after all tests are updated/reviewed.

CC @Fraser999

Reply:
The tests are now updated.